core/trino-spi/src/main/java/io/trino/spi/PageBuilder.java

```diff
@@ -98,6 +98,19 @@ public void reset()
         }
     }
 
+    public void resetExact()
```

Comment:
90%* of method names are ambiguous (see no further than `newBlockBuilderLike`).
I think keeping the method without `expectedEntries` param is better as we don't need the extended functionality of resetting `PageBuilder` to any position count.

* made up stat

Reply:
Definitely needs a javadoc
Alternatively, how about a `PageBuilder newPageBuilderLike(int expectedEntries)` method instead ?

Reply:
to avoid the Javadoc but also avoid exposing unnecessary functionality I added **private**
`private void reset(int expectedEntries)` and **public** resetExact with an extremely straightforward implementation that does not need documentation.

Reply:
Since it's public method in SPI and the name "resetExact" doesn't easily convey what this is about, we should still add a javadoc explaining what this method is about. Simplifying the implementation doesn't take away the need for javadoc unless the name of the method is also simple enough to make explicit documentation unnecessary.
How about a `PageBuilder newPageBuilderLike(int expectedEntries)` method instead ? That probably wouldn't need a javadoc

Reply:
I added a comment

> How about a PageBuilder newPageBuilderLike(int expectedEntries) method instead 

As I mentioned above, we don't need additional functionality with an arbitrary number of `expectedEntries` to be publicly available. On top of that, I assume you want `newPageBuilderLike` to create a new `PageBuilder` instance and that would require additional changes to the constructor.

Reply:
> 90%* of method names are ambiguous (see no further than newBlockBuilderLike).

We shouldn't add even more ambiguity.

`reset(int expectedEntries)` is explicit about it's semantics and doesn't require extra Javadoc really.

`resetExact` is an SPI method implementation for your particular use-case, while `reset(int expectedEntries)` corresponds more of a `PageBuilder` constructor that accepts `initialExpectedEntries`

Reply:
`reset(int expectedEntries)` may correspond to the `PageBuilder` constructor but it's not needed, why add it to the public API?

How about instead of inlining `resetExact` we change the name to be less ambiguous? The approach of inlining method calls because we don't like the method name is not good long term.

Reply:
I'm not sure you can inline it TBH due to page builder status

Reply:
ping https://github.com/trinodb/trino/pull/12512/commits/9e984abd9fdf112c68a04119e4146b83c870af00#r950126334

Reply:
@sopel39 see https://github.com/trinodb/trino/pull/12512/files#r950167250
to sum up. we don't need `reset(int expectedEntries)` public API, we need `resetExact`. If the name is misleading please propose a better name but I stand by the fact that `resetExact` captures the semantics we need and `reset(int expectedEntries)` adds complexity.

Reply:
in order to move forward with the PR I dropped the `resetExact` and made `void reset(int expectedEntries)` public.